### PR TITLE
Add focus-hover and group-focus-hover variants

### PR DESCRIPTION
**Pull request by mzlock**

### CODE REVIEW

File: plugins/focus-hover/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = () => ({ addVariant, e }) => {
```
Comment (jeremyfrank) [inline plugins/focus-hover/index.js]: Wrap with new plugin function.

File: plugins/group-focus-hover/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module.exports = () => ({ addVariant, e }) => {
```
Comment (jeremyfrank) [inline plugins/group-focus-hover/index.js]: Wrap with new plugin function.

File: plugins/group-focus-hover/index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  addVariant('group-focus-hover', ({ modifySelectors, separator }) => {
+    modifySelectors(({ className }) => {
+      return `
+        .group:focus .${e(`group-focus-hover${separator}${className}`)},
+        .group:hover .${e(`group-focus-hover${separator}${className}`)}
+      `
+    })
+  })
```
Comment (jeremyfrank) [inline plugins/group-focus-hover/index.js:11]: Thoughts on following the pattern used by Tailwind core for `group-hover`? That's what I used for parent-open and parent-expanded. https://github.com/tailwindcss/tailwindcss/blob/6343d1ec7c1d6eb21cce8b2ded02bc5f700e99f6/src/lib/substituteVariantsAtRules.js#L26-L38

**Review by jeremyfrank — APPROVED** [review]
Just two spots for wrapping in the plugin function. You can see examples of where Greg added this in the other plugins. And one other question about a pattern from TW core.
https://github.com/tailwindcss/tailwindcss/releases/tag/v1.2.0

File: plugins/group-focus-hover/index.js (hunk 3 above)
Comment (gregkohn) [reply plugins/group-focus-hover/index.js:11]: I would love to know if there's a difference if `modifySelectors()` works as-is.
The docs seem to encourage the more complex approach only if you can't achieve it with `modifySelectors()`?
https://tailwindcss.com/docs/plugins#basic-variants

Comment (mzlock) [reply plugins/group-focus-hover/index.js:11]: I have no thoughts on one versus the other here. Let me know if I should change it to match our preference? 🤷

**Comment by gregkohn** [comment]
One question I have for yall -- how do we feel about `hocus` or something along those lines that's shorter?

**Comment by mzlock** [comment]
hocus-pocus

**Comment by jeremyfrank** [comment]
Ha, hocus, there's actually another plugin that names it that way.
https://github.com/benface/tailwindcss-interaction-variants

File: plugins/group-focus-hover/index.js (hunk 3 above)
Comment (jeremyfrank) [reply plugins/group-focus-hover/index.js:11]: 🤷‍♂️ Not sure what the difference is. I just copied/pasted/modified the TW source for `group-hover` when I made my parent-expanded and parent-open plugins. I'm fine with either approach.

**Comment by mzlock** [comment]
Thoughts on using that plugin instead and ditching my version? Might be nice if it's more full-featured. And I'm sure the `hocus` variants work the same way

**Comment by jeremyfrank** [comment]
Hmm, yeah might make sense to use that plugin instead since it seems pretty fully featured. The individual group-hocus variant outputs pretty much the same way too.

**Comment by mzlock** [comment]
Closing in favor of adopting https://github.com/benface/tailwindcss-interaction-variants

**Pull request closed without merging**